sphinx/writers/latex.py

```diff
@@ -473,6 +473,7 @@ def __init__(self, document, builder):
         self.in_term = 0
         self.needs_linetrimming = 0
         self.in_minipage = 0
+        self.in_box = 0
```

Comment:
Although indeed the sphinx.sty LaTeX macros use "sphinxlightbox" and "sphinxheavybox" terminology, in the former case (note, hint, important, tip), the contents are not encapsulated in a box, and there was no crash in those cases. However it does make sense to prohibit floating figures also for them. Perhaps `no_latex_floats` (or `no_floating_figures` as Sphinx uses LaTeX floats only for figures) would be better than `in_box`.